src/poetry/repositories/pool.py

```diff
@@ -63,133 +82,62 @@ def add_repository(
         Adds a repository to the pool.
         """
         repository_name = repository.name.lower()
-        if repository_name in self._lookup:
-            raise ValueError(f"{repository_name} already added")
-
-        if default:
-            if self.has_default():
-                raise ValueError("Only one repository can be the default")
+        if self.has_repository(repository_name):
+            raise ValueError(
+                "A repository with (case-independent) name"
+                f" {repository_name} was already added"
```

Comment:
```suggestion
                "A repository with name {repository_name}"
                f" was already added."
```

I think we can drop the "case-independent" bit as we're printing the name we're using in the exception.